hlint.cabal

```diff
@@ -53,7 +53,7 @@ flag gpl
     description: Use GPL libraries, specifically hscolour
 
 flag ghc-lib
-  default: False
+  default: True
```

Comment:
Good call - this seems like the right time to flip this default

Reply:
jolly good. i've taken the liberty of pushing the change to `ci.yml` then that bumps the macOS compiler to ghc-9.2.3 .

Reply:
@ndmitchell @shayne-fletcher what's the rationale for this change?

For `ormolu` I was proposing to use `ghc` by default when possible (https://github.com/tweag/ormolu/pull/948). Are there any potential drawbacks with that?

Reply:
@sol the motivation is discussed in https://github.com/ndmitchell/hlint/issues/1376. i don't know of any drawbacks. you may find [this blog post](https://blog.shaynefletcher.org/2021/02/configuring-cabal-build-flags.html) helpful.